pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/TeiXml.scala

```diff
@@ -95,8 +100,76 @@ class TeiXml(val xml: Elem) extends Logging {
     }
   }
 
-  private def getId: Result[String] = TeiOps.getIdFrom(xml)
+  /**
+    * The origin tag contains information about where and when
+    * the manuscript was written. This is an example:
+    *  <history>
+    *      <origin>
+    *          <origPlace>
+    *              <country><!-- insert --></country>,
+    *              <region><!-- insert --></region>,
+    *              <settlement><!-- insert --></settlement>,
+    *              <orgName><!-- insert --></orgName>
+    *          </origPlace>
+    *          <origDate calendar=""><!-- insert --></origDate>
+    *      </origin>
+    *  </history>
+    *
+    */
+  private def origin: Result[List[ProductionEvent[Unminted]]] = {
+    val origin = xml \\ "history" \ "origin"
+    val origPlace = origin \ "origPlace"
+    val country = (origPlace \ "country").text.trim
+    val region = (origPlace \ "region").text.trim
+    val settlement = (origPlace \ "settlement").text.trim
+    val organisation = (origPlace \ "orgName").text.trim
+    val date = parseDate(origin)
+    val place =
+      List(country, region, settlement).filterNot(_.isEmpty).mkString(", ")
+    val agents =
+      if (organisation.isEmpty) Nil else List(Organisation(organisation))
+    val places = if (place.isEmpty) Nil else List(Place(place))
+    val dates = if (date.isEmpty) Nil else List(ParsedPeriod(date))
+    val label = List(place, date).filterNot(_.isEmpty).mkString(", ")
+    (agents, places, dates) match {
+      case (Nil, Nil, Nil) => Right(Nil)
+      case _ =>
+        Right(
+          List(
+            ProductionEvent(
+              label = label,
+              places = places,
+              agents = agents,
+              dates = dates
+            )))
+    }
+
+  }
+
+  /**
+    * Dates are in a origDate tag and can be in different calendars,
+    * so we need to look for the one in the gregorian calendar.
+    * Also, sometimes the date can contain notes, as in this example, so we need to strip them:
+    * <origDate calendar="Gregorian">ca.1732-63AD
+    *  <note>from watermarks</note>
+    * </origDate>
+    */
+  private def parseDate(origin: NodeSeq) = {
```

Comment:
Would any of [our existing parsers](https://github.com/wellcomecollection/catalogue-pipeline/tree/main/pipeline/transformer/transformer_common/src/main/scala/weco/pipeline/transformer/parse) help?

Reply:
ah thanks, I didn't see those

Reply:
Actually, I am using one of our parsers above. This function is more about extracting the string with the date (perhaps the name of the function could be better)